src/Integration.UnitTests/Helpers/ToolWindowServiceTests.cs

```diff
@@ -40,29 +39,35 @@ public class ToolWindowServiceTests
 
         [TestMethod]
         public void MefCtor_CheckIsExported()
-        {
-            // Set up a service provider with the required service
-            var serviceProviderMock = new Mock<IServiceProvider>();
-            serviceProviderMock.Setup(x => x.GetService(typeof(SVsUIShell))).Returns(Mock.Of<IVsUIShell>());
+            => MefTestHelpers.CheckTypeCanBeImported<ToolWindowService, IToolWindowService>(
+                MefTestHelpers.CreateExport<IVsUIServiceOperation>());
+
+        [TestMethod]
+        public void MefCtor_CheckIsSingleton()
+            => MefTestHelpers.CheckIsSingletonMefComponent<ToolWindowService>();
 
-            var serviceProviderExport = MefTestHelpers.CreateExport<SVsServiceProvider>(serviceProviderMock.Object);
+        [TestMethod]
+        public void MefCtor_DoesNotCallAnyServices()
+        {
+            var serviceOp = new Mock<IVsUIServiceOperation>();
 
-            // Act and Assert
-            MefTestHelpers.CheckTypeCanBeImported<ToolWindowService, IToolWindowService>(serviceProviderExport);
+            _ = CreateTestSubject(serviceOp.Object);
 
-            serviceProviderMock.VerifyAll();
+            // The MEF constructor should be free-threaded, which it will be if
+            // it doesn't make any external calls.
+            serviceOp.Invocations.Should().BeEmpty();
         }
 
         [TestMethod]
         public void Show_FindWindowFailed_NoExceptionAndFrameNotShown()
         {
-            var serviceProviderMock = new Mock<IServiceProvider>();
             var uiShellMock = new Mock<IVsUIShell>();
             var frameMock = new Mock<IVsWindowFrame>();
+            SetupFindToolWindow(uiShellMock, VSConstants.E_OUTOFMEMORY, ValidToolWindowId, frameMock.Object);
 
-            SetupFindToolWindow(serviceProviderMock, uiShellMock, VSConstants.E_OUTOFMEMORY, ValidToolWindowId, frameMock.Object);
+            var serviceOp = CreateServiceOperation(uiShellMock.Object);
 
-            var testSubject = new ToolWindowService(serviceProviderMock.Object);
```

Comment:
Again, I added the conventional `CreateTestSubject` method, even though there is only one parameter to the constructor at the moment.